Code/Editor/ViewManager.cpp

```diff
@@ -210,8 +212,50 @@ void CViewManager::Cycle2DViewport()
 //////////////////////////////////////////////////////////////////////////
 CViewport* CViewManager::GetViewportAtPoint(const QPoint& point) const
 {
-    QWidget* widget = QApplication::widgetAt(point);
-    return qobject_cast<QtViewport*>(widget);
+    for (auto& vp : m_viewports)
+    {
+        if (!vp || !vp->widget())
+        {
+            continue;
+        }
+        if (vp->widget()->rect().contains(vp->widget()->mapFromGlobal(point)))
+        {
+            return vp;
+        }
+    }
+
+    return nullptr;
```

Comment:
minor: You could also use `find_if`
```suggestion
    auto vpIter = AZStd::find_if(m_viewports.begin(), m_viewports.end(), [&point](CViewport* vp)->bool {
        auto* widget = vp->widget();
        return widget && widget->rect().contains(widget->mapFromGlobal(point));
    });
    CViewport* vp = (vpIter==m_viewports.end())?nullptr:*vpIter;
    return vp;
```